File: frontend/src/api/cashflow.js
/**
 * Cash-flow outlook — the forward-looking counterpart to Commitments.
 *
 * The payload is an estimate and carries its own confidence: when
 * `discretionary_basis.confidence` is "none" the typical-spending figure is
 * omitted and `projection_incomplete` is true, so callers say so rather than
 * presenting a recurring-only net as the whole picture.
 */
import axios from 'axios';

const API = process.env.REACT_APP_BACKEND_URL || '';

export const getCashflowProjection = (horizonDays = 30) =>
  axios.get(`${API}/api/cashflow/projection`, { params: { horizon_days: horizonDays } });

File: backend/routers/cashflow.py
"""Cash-flow outlook — a thin read over ``analytics.project_cashflow``.

No response model: the payload is an estimate whose shape the service owns,
and the projection carries its own confidence, so pinning it twice would only
let the two drift.
"""
from typing import Any, Dict

from fastapi import APIRouter, Query

from analytics import project_cashflow

router = APIRouter()


@router.get("/cashflow/projection")
async def get_cashflow_projection(
    horizon_days: int = Query(30, ge=1, le=180),
) -> Dict[str, Any]:
    """Income in, bills out, typical spending out, projected net."""
    return project_cashflow(horizon_days=horizon_days)
